src/main/scala/firrtl/backends/verilog/VerilogEmitter.scala

```diff
@@ -1200,13 +1223,19 @@ class VerilogEmitter extends SeqTransform with Emitter {
         for (x <- initials) emit(Seq(tab, x))
         for (x <- asyncInitials) emit(Seq(tab, x))
         emit(Seq("  `endif // RANDOMIZE"))
-        for (x <- memoryInitials) emit(Seq(tab, x))
+
+        if (emissionOptions.emitMemoryInitAsNoSynth.get) {
+          for (x <- memoryInitials) emit(Seq(tab, x))
+        }
```

Comment:
You definitely don't want to do a `.get`, you'll throw an exception if neither annotation is provided.

Reply:
I see now that you're providing a default, let's just drop the `Option` on the return.

Reply:
Done!